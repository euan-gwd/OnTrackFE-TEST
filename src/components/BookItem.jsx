import React from 'react';

const BookItem = ({ book }) => (
  <div>
    <p>
      Title: <span>{book.book_title}</span>
    </p>
    <p>
      Author(s):
      <span>
        {book.book_author.map((author, idx) => (
          <span key={idx}>{` ${author} `}</span>
        ))}
      </span>
    </p>
    <p>
      Publication year: <span>{book.book_publication_year}</span>
    </p>
    <p>
      Publication country : <span>{book.book_publication_country}</span>
    </p>
    <p>
      Publication city : <span>{book.book_publication_city}</span>
    </p>
    <p>
      Total pages: <span>{book.book_pages}</span>
    </p>
    <p>
      Id: <span>{book.id} </span>
    </p>
  </div>
);

export default BookItem;
