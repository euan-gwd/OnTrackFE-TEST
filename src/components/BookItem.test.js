import React from 'react';
import { render } from '@testing-library/react';
import BookItem from './BookItem';

test('renders as expected', () => {
  const book = {
    id: 1,
    book_author: ['mocked'],
    book_title: 'mocked',
    book_publication_year: 2020,
    book_publication_country: 'mocked',
    book_publication_city: 'mocked',
    book_pages: 104
  };

  const comp = render(<BookItem book={book} />);
  expect(comp).toMatchSnapshot();
});
