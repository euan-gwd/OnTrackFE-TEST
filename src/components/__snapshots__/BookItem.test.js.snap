// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          Title: 
          <span>
            mocked
          </span>
        </p>
        <p>
          Author(s):
          <span>
            <span>
               mocked 
            </span>
          </span>
        </p>
        <p>
          Publication year: 
          <span>
            2020
          </span>
        </p>
        <p>
          Publication country : 
          <span>
            mocked
          </span>
        </p>
        <p>
          Publication city : 
          <span>
            mocked
          </span>
        </p>
        <p>
          Total pages: 
          <span>
            104
          </span>
        </p>
        <p>
          Id: 
          <span>
            1
             
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        Title: 
        <span>
          mocked
        </span>
      </p>
      <p>
        Author(s):
        <span>
          <span>
             mocked 
          </span>
        </span>
      </p>
      <p>
        Publication year: 
        <span>
          2020
        </span>
      </p>
      <p>
        Publication country : 
        <span>
          mocked
        </span>
      </p>
      <p>
        Publication city : 
        <span>
          mocked
        </span>
      </p>
      <p>
        Total pages: 
        <span>
          104
        </span>
      </p>
      <p>
        Id: 
        <span>
          1
           
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
